### PR TITLE
[BEAM-2532] add a Serializable TableSchema Supplier in BigQuerySourceBase

### PR DESCRIPTION
**Pull request by nevillelyh**
Verified that with the fix, loading a 35GB BigQuery table with 10 n1-standard-1 workers is 4x faster than without (~10min vs 40min).

### CODE REVIEW

**Comment by reuvenlax** [comment]
I suspect you'll have to add a FindBugs exception. See https://github.com/apache/beam/blob/996e35c1de8310a265c890cf8ca8e92416b3d16b/sdks/java/build-tools/src/main/resources/beam/findbugs-filter.xml#L67 for an example.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQuerySourceBase.java [hunk 1]
```diff
@@ -168,10 +170,12 @@ public void validate() {
 
     SerializableFunction<GenericRecord, TableRow> function =
         new SerializableFunction<GenericRecord, TableRow>() {
+          private Supplier<TableSchema> schema = new SerializableTableSchemaSupplier(
```
Comment (jkff) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQuerySourceBase.java]: Hmm, can this be simplified by making this a `Suppliers.memoize(Suppliers.compose(BigQueryHelpers::fromJsonString, Suppliers.ofInstance(jsonSchema)))` ? (except the lambda has to be Java 7)

using methods from http://google.github.io/guava/releases/22.0/api/docs/com/google/common/base/Suppliers.html

**Comment by jkff** [comment]
R: @jkff

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQuerySourceBase.java (hunk 1 above)
Comment (nevillelyh) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQuerySourceBase.java]: Done

**Pull request closed without merging**